go/cmd/dolt/commands/tag.go

```diff
@@ -112,81 +136,113 @@ func (cmd TagCmd) Exec(ctx context.Context, commandStr string, args []string, dE
 	} else if len(apr.Args) > 2 {
 		verr = errhand.BuildDError("create tag takes at most two args").Build()
 	} else {
-		props, err := getTagProps(dEnv, apr)
-		if err != nil {
-			verr = errhand.BuildDError("failed to get tag props").AddCause(err).Build()
-			return HandleVErrAndExitCode(verr, usage)
-		}
 		tagName := apr.Arg(0)
 		startPoint := "head"
 		if len(apr.Args) > 1 {
 			startPoint = apr.Arg(1)
 		}
-		err = actions.CreateTag(ctx, dEnv, tagName, startPoint, props)
+		msg, _ := apr.GetValue(cli.MessageArg)
+		author, _ := apr.GetValue(cli.AuthorParam)
+
+		err = createTag(queryist, sqlCtx, tagName, startPoint, msg, author)
+
 		if err != nil {
 			verr = errhand.BuildDError("failed to create tag").AddCause(err).Build()
 		}
 	}
-
 	return HandleVErrAndExitCode(verr, usage)
 }
 
-func getTagProps(dEnv *env.DoltEnv, apr *argparser.ArgParseResults) (props actions.TagProps, err error) {
-	var name, email string
-	if authorStr, ok := apr.GetValue(cli.AuthorParam); ok {
-		name, email, err = cli.ParseAuthor(authorStr)
+func createTag(queryist cli.Queryist, sqlCtx *sql.Context, tagName, startPoint, message, author string) error {
+	var query string
+	var params []interface{}
+	if len(message) == 0 {
+		if len(author) == 0 {
+			query = "call dolt_tag(?, ?)"
+			params = []interface{}{tagName, startPoint}
+		} else {
+			query = "call dolt_tag(?, ?, '--author', ?)"
+			params = []interface{}{tagName, startPoint, author}
+		}
 	} else {
-		name, email, err = env.GetNameAndEmail(dEnv.Config)
+		if len(author) == 0 {
+			query = "call dolt_tag('-m', ?, ?, ?)"
```

Comment:
You should always pass in an author. The stored procedure doesn't have the local user information available to it, so it puts in root@dolt (or something like that) as the author.

Reply:
Hmm, that's not what I'm observing. This BATS test just succeeded for me. I'm specifically checking for local/remote execution, not specifying an author, and verifying that the author is `Bats Tests <bats@email.fake>`:

```sh
  # add tag without message
  run dolt --verbose-engine-setup tag v1
  [ $status -eq 0 ]
  [[ $output =~ "verbose: starting remote mode" ]] || false

  # list tags and check new tag is present
  run dolt tag
  [ $status -eq 0 ]
  [[ $output =~ "v1" ]] || false

  # list tags with verbose flag and check new tag is present
  run dolt --verbose-engine-setup tag --verbose
  [ $status -eq 0 ]
  [[ $output =~ "v1"$'\t'"$headCommit" ]] || false
  [[ $output =~ "Tagger: Bats Tests <bats@email.fake>" ]] || false
  [[ $output =~ "verbose: starting remote mode" ]] || false
```

Reply:
Huh. Interesting. I'm convinced by the test :)